arch/risc-v/src/common/riscv_pmp.c

```diff
@@ -318,19 +318,17 @@ static uintptr_t pmp_read_addr(uintptr_t region)
  *
  * Input Parameters:
  *   val  - Value to decode.
- *   base - Base out.
  *   size - Size out.
  *
  * Returned Value:
- *   None
+ *   Base address
  *
  ****************************************************************************/
 
-static void pmp_napot_decode(uintptr_t val, uintptr_t * base,
-                             uintptr_t * size)
+static uintptr_t pmp_napot_decode(uintptr_t val, uintptr_t * size)
 {
-  uint64_t mask = (uint64_t)(-1) >> 1;
-  uint64_t pot  = __riscv_xlen + 2;
+  uintptr_t mask = (uintptr_t)(-1) >> 1;
+  uintptr_t pot  = __riscv_xlen + 2;
```

Comment:
Let's remove add `__riscv_xlen` references and switch to `CONFIG_ARCH_RV32`. Here for example we can go with
```suggestion
  uintptr_t pot  = sizeof(uintptr_t) + 2;
```

Reply:
Can you elaborate a bit why __riscv_xlen is undesireable ? As far as I know the compiler provides this definition which can be trusted as long as it is invoked with the correct target type.
I can of course avoid using it if there is a clear motivation to do so.

Just a note sizeof(uintptr_t) != __riscv_xlen. It is the machine bit width I want here, not the size of uintptr_t in octets.

Reply:
Yes. I made a mistake. It should be `sizeof(uintptr_t) * CHAR_BIT` you are right

Reply:
I'll remove the reference to riscv_xlen